apps/ai-content-generator/src/components/app/dialog/common-generator/output/output-text-panels/generated-text-panel/GeneratedTextPanel.spec.tsx

```diff
@@ -0,0 +1,71 @@
+import { render, renderHook } from '@testing-library/react';
+import { describe, expect, it, vi } from 'vitest';
+import { AIMock, mockCma, MockSdk } from '@test/mocks';
+import GeneratedTextPanel from './GeneratedTextPanel';
+import useAI from '@hooks/dialog/useAI';
+import { Tabs } from '@contentful/f36-components';
+import { OutputTab } from '../../Output';
+
+const mockSdk = new MockSdk();
+const sdk = mockSdk.sdk;
+
+vi.mock('@contentful/react-apps-toolkit', () => ({
+  useSDK: () => sdk,
+  useCMA: () => mockCma,
+}));
+
+vi.mock('@utils/aiApi', () => AIMock);
+
+describe('GeneratedTextPanel', () => {
+  it('renders', () => {
+    const hook = renderHook(() => useAI());
+    const { getByText, unmount, rerender } = render(
+      <Tabs currentTab={OutputTab.GENERATED_TEXT}>
+        <GeneratedTextPanel
+          generate={() => {}}
+          ai={hook.result.current}
+          outputFieldValidation={null}
+          apply={() => {}}
+        />
+      </Tabs>
+    );
+    expect(getByText('Regenerate')).toBeTruthy();
+    hook.result.current.output = 'test';
+
+    rerender(
+      <Tabs currentTab={OutputTab.GENERATED_TEXT}>
+        <GeneratedTextPanel
+          generate={() => {}}
+          ai={hook.result.current}
+          outputFieldValidation={null}
+          apply={() => {}}
+        />
+      </Tabs>
+    );
+
+    expect(getByText('4 characters')).toBeTruthy();
```

Comment:
Where is this text `4 characters` coming from? It's not clear to me in the test if this is some default text.

Reply:
The 4 chars are coming from line 33. where I'm setting the `{ output }` value from `useAI()` to "text"

Reply:
Thank you! Totally missed that! 😅